src/test/java/edu/pdx/cs/ClientTest.java

```diff
@@ -48,30 +43,61 @@ public void testLoginMethod() throws IOException {
         String password = "katara";
 
         Client client = new Client();
-        client.connect("localhost", 8080);
+        client.connect(HOSTNAME, Integer.parseInt(PORT));
 
         assertThat(client.login(username, password), equalTo(true));
     }
 
-    @After
-    public void teardown() {
-        server.stop();
+    @Test
+    public void testListRemoteFiles() throws IOException{
+        Client client = new Client();
+        client.connect(HOSTNAME, Integer.parseInt(PORT));
+        client.login("aang", "katara");
+
+        FTPFile[] files = client.listRemoteFiles();
+
+        assertThat(files.length, equalTo(2));
+        assertThat(files[0].isDirectory(), equalTo(false));
+        assertThat(files[0].getName(), containsString("love_note"));
+        assertThat(files[1].isDirectory(), equalTo(false));
+        assertThat(files[1].getName(), containsString("momo.gif"));
+    }
+
+    @Test
+    public void testListRemoteDirectories() throws IOException {
+        Client client = new Client();
+        client.connect(HOSTNAME, Integer.parseInt(PORT));
+        client.login("aang", "katara");
+
+        FTPFile[] directories = client.listRemoteDirectories();
+
+        assertThat(directories.length, equalTo(1));
+        assertThat(directories[0].isDirectory(), equalTo(true));
+        assertThat(directories[0].getName(), containsString("testDir"));
     }
```

Comment:
It's probably best to abstract out tests out into their own class or classes. Whatever makes sense. Otherwise ClientTest will get really bloated.

Reply:
Can do.